operator/v1/types_console.go

```diff
@@ -143,8 +143,94 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeType is the value of the CustomLogo Theme Type field that determines the theme mode in the console UI.
+type ThemeType string
+
+// ThemeType values
+const (
+	// ThemeTypeDark represents the dark theme for the console UI.
+	ThemeTypeDark ThemeType = "Dark"
+
+	// ThemeTypeLight represents the light theme for the console UI.
+	ThemeTypeLight ThemeType = "Light"
+
+	// ThemeTypeDefault represents the default theme for all unspecified themes.
+	ThemeTypeDefault ThemeType = "Default"
+)
+
+// LogoType is the value of the CustomLogo Type field that determines if the logo is for the masthead or the favicon in the console UI.
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// Theme defines a theme type for the console UI.
+type Theme struct {
+	// type is used to specify what theme(s) a logo will apply to in the console UI.
+	// type is a required field that allows values of Dark, Light, and Default.
+	// When set to Dark, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Dark theme.
+	// When set to Light, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Light theme.
+	// When set to Default, the logo file referenced in the 'file' field will be used for any themes types that do not have an explicit configuration.
+	// +kubebuilder:validation:Enum="Dark";"Light";"Default"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is used by the console to locate the specified file containing a custom logo.
```

Comment:
Would there be a way to name this so that it's obvious that we are referring to a configmap when serialized as well?

I'm concerned that `file.name` vs `file.key` might be confusing, the file name is typically what you would call the value of the key field in this instance

Reply:
It's not a configmap file though is it? We are storing a logo file within a configmap.

Might be worth seeing if we can see prior examples of this kind of reference in other places

Reply:
+1 for a simple `FileReference` placed in the operator types, which we have ownership over.

Reply:
Do you think there will ever be a different way to source the logo apart from configmaps? I'm thinking maybe source, and remove the file bit altogether
```
sourceFrom:
  configMap:
    name: blah
    key: something
```

Reply:
I'm a fan of the `source`/`sourceFrom` with a discriminated union for different source types. When I was on the OLM team we did something like that in OLMv1's `ClusterCatalog` type with a single allowed source type of `Image`. I don't know if it has progressed further but the idea was that we were thinking to add future sourcing options. 

I think that same idea could fit well here and allow you to expand how customers can tell console to fetch their custom logos.

Reply:
Would it be possible in the future for them to just provide a logo URL perhaps?